### ISSUE TITLE
[eWeLink] Unable to reach eWeLink. Retrying in 30 seconds.

### ISSUE THREAD

**Issue by marekzrostlik** [bwp91/homebridge-ewelink#192]
Hello, I was running this homebridge-ewelink plugin without any problems for few fews. A then suddenly the plugin is irregularly having issues to connect the API host. These are the lines from the Homebridge log:
[eWeLink] Sending HTTP getHost() request.
[eWeLink] Unable to reach eWeLink. Retrying in 30 seconds.

Is there a way, how I could verify the availability of the host manually through URL in my webbrowser or so? Or sniff IPs and ports to verify what is going on there?

For a full picture, this is my config. Username/password masked:
{
            "name": "eWeLink",
            "countryCode": 420,
            "username": "XXXXXXXX",
            "password": "XXXXXXXX",
            "mode": "auto",
            "debug": true,
            "disablePlugin": false,
            "ipOverride": [
                {
                    "deviceID": "10010a5d32",
                    "deviceIP": "192.168.1.141"
                }
            ],
            "platform": "eWeLink"
        }

Thanks for your guidance, M.

**Comment by bwp91** [comment]
Hi @marekzrostlik 

The plugin will automatically retry to send the http request if the initial response code from the eWeLink server is one of the following: `ENOTFOUND`, `ETIMEDOUT`, `EAI_AGAIN`

Which are normally due to:

* temporary eWeLink server issues
* internet issues you are experiencing yourself

The reason why the plugin **only** retries on these status codes is that they are normally intermittent so eventually they will go away without further user intervention. For other status codes there would normally need to be some action by the user (and the plugin won't reattempt automatically).

I'm not sure of how well you can read javascript, but here is a link to the function that you're facing trouble with:

https://github.com/bwp91/homebridge-ewelink/blob/e1b925157cd103ee53add591fae5c201f1b3dd2e/lib/connection/http.js#L26

On line 56 you can see the URL that the plugin accesses, and on the previous few lines the required headers that are needed with the request.

In the next version of the plugin I will show the returned http code in each case in the logs so that you can see exactly which code is being returned.

Hope this helps

**Comment by marekzrostlik** [comment]
Hello Ben,

Thanks for the quick reply!

It seems that in my case the roubles are related to the network routing. Possibly assymetric routing paths back and forth. My router has two internet trunks. And possibly ewelink server can adapt to this. I will elaborate on this first.

**Comment by bwp91** [comment]
In v4.7.0 you should be able to see the error code in the log if/when you receive the error.

**Comment by marekzrostlik** [comment]
Sorry, for not responding quicker. I did multiple tests and they took some time.

Yes, your answers were very useful. The problem in my case is an assymetric routing path, thus I am now consulting network admin.

This issue can be closed.

**Comment by marekzrostlik** [comment]
Hello @bwp91 ,

I am following our previous exchange above https://github.com/bwp91/homebridge-ewelink/issues/192#issuecomment-774924668.

Network engineer reworked router config in order to remove assymetric routing. Nevertheless Homebridge log is still showing error message that it cant reach ewelink server. May be that the issue is in my "countryCode": 420 defining my region and thus ewelink server to be accessed? I am from the Czech Republic. Am I supposed to specify country code as "420", "42" or "cz"?

**Comment by bwp91** [comment]
Try `"countryCode": "420”` does this work?

**Comment by marekzrostlik** [comment]
Thanks. Have tried "420" but didnt work. Then tried "42" and it didnt work either.
Have no clue what the issue could be. What is the IP of ewelink server? To double check I can try to ping or tracert it.

It is really confusing, because I am almost sure that the link this plugin worked just fine some time ago. Obviously with its different version. And that time in my opinion "420" was in the config. Then since I updated the plugin, the issues with unreachability of ewelink server started I never disappeared since then. I elaborated all combinations of config settings. Nothing helped so far.
